chatter.js

```diff
@@ -1,12 +1,211 @@
+
+/**
+ * @summary Retrieves nickname of user.
+ * @locus Server
+ * @param {object} meteor user.
+ * @returns {string} the nickname of the user.
+ */
+function getNickname(user) {
+  const nickPath = Chatter.options.nickProperty;
+  const nick = nickPath.split('.').reduce((prevVal, el) => prevVal[el] , user);
+
+  return nick;
+}
+
+/**
+ * @summary Adds user to Chatter and defines its privileges
+ * @locus Server
+ * @param params Information about the new user.
+ * @param {string} params.userId Meteor user id.
+ * @param {string} params.userType Defines the restricitons on the new user. Can either be set to "admin" or "standard".
+ * @returns {string} chatterId
+ */
+Chatter.addUser = function(params) {
+  check(params, {
+    userId: String,
+    admin: Match.Optional(Match.OneOf(Boolean, String, undefined))
+  });
+
+  const {userId, userType} = params;
+
+  const user = Meteor.users.findOne(userId);
+
+  if (!user) {
+    throw new Meteor.Error("user-does-not-exists", "user id provided is not correct");
+  }
+
+  if (user.profile.isChatterUser) {
+    throw new Meteor.Error("user-already-exists", "user has already been added to chatter");
+  }
+
+  const isAdmin = isChatterAdmin ? true : false;
+  const avatarURL = `http://api.adorable.io/avatars/${user.username}`;
+
+  Meteor.users.update(
+    {_id: userId},
+    { $set: {
+      "profile.isChatterUser": true,
+      "profile.isChatterAdmin": isAdmin,
+      "profile.chatterNickname": user.username,
+      "profile.chatterAvatar": avatarURL,
     }
+  });
+
+  return userId;
 };
 
+/**
+ * @summary Removes user from Chatter
+ * @locus Server
+ * @param params Information about the new user.
+ * @param {string} params.userId Meteor user id.
+ * @param {string} params.userType Defines the restricitons on the new user. Can either be set to "admin" or "standard".
+ * @returns {string} chatterId
+ */
+Chatter.removeUser = function(params) {
+  check(params, {
+    userId: String
+  });
 
-Chatter.configure = function (opts) {
-    _.extend(this.options, opts);
+  const {userId} = params;
+  const user = Meteor.users.findOne(userId);
+
+  if (!user) {
+    throw new Meteor.Error("user-does-not-exists", "user id provided is not correct");
+  }
+
+  if (!user.profile.isChatterUser) {
+    throw new Meteor.Error("user-is-not-chatter-user", "user is not a chatter user");
+  }
+
+  Meteor.users.update({_id: userId}, { $set: {"profile.isChatterUser": false} });
+  return userId;
 };
 
+/**
+ * @summary Adds empty room to Chatter.
+ * @locus Server
+ * @param {string} params Information about the new room.
+ * @param {string} params.name The name of the room.
+ * @param {string} params.description The description of the room.
+ * @returns {string} roomId
+ */
+Chatter.addRoom = function(params) {
+  check(params, {
+    name: String,
+    description: String
+  });
+
+  const {name, description} = params;
+  const room = new Chatter.Room({
+    name,
+    description
+  })
+
+  if (room.validate()) {
```

Comment:
The save and validate functionality are surprisingly kept separate on purpose: https://github.com/jagi/meteor-astronomy/issues/48 Apparently there are some cases where you want to save without validating, (fetching data from server and saving it again on database) What I could do would be something like the below to automatise it

Astro.eventManager.on('beforesave', function(e) {
  if (!e.target.validate()) {
      e.target.throwValidationException();
  }
});